pkg/fleet/installer/installer.go

```diff
@@ -524,6 +524,11 @@ func (i *installerImpl) InstallConfigExperiment(ctx context.Context, pkg string,
 		)
 	}
 
+	// HACK: close so package can be updated as watchdog runs
+	if pkg == packageDatadogAgent && runtime.GOOS == "windows" {
```

Comment:
Not sure I follow this hack, I would have expected the hook to return quickly after launching the background command and close the DB ?

Reply:
i copied it over from update's `InstallExperiment`, so I'm not sure about the origin of the `HACK` label, I left the comment as-is to ensure they'd be updated at the same time.

I'd rather close the db ahead of time to prevent any races. I think there's some general work to be done since the db is always opened by the constructor but many of the installer subcommands never actually use it.

Reply:
oh this change ended up in the diff when github auto rebased it after this PR passed the merge queue. sorry for the confusion.
https://github.com/DataDog/datadog-agent/pull/37494/files#diff-5097e32b945d38543ad437011a6c5e9e11312ede815b9788d7788164c871627a